packages/nextjs/app/debug/_components/contract/utilsDisplay.tsx

```diff
@@ -24,33 +24,87 @@ export const displayTxResult = (
   if (typeof displayContent === "bigint") {
     try {
       const asNumber = Number(displayContent);
-      if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
-        return asNumber;
+      if (asText) {
+        if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
+          return asNumber;
+        } else {
+          return "Ξ" + formatEther(displayContent);
+        }
       } else {
-        return "Ξ" + formatEther(displayContent);
+        return <NumberDisplay value={displayContent} />;
       }
     } catch (e) {
       return "Ξ" + formatEther(displayContent);
     }
   }
 
-  if (typeof displayContent === "string" && isAddress(displayContent)) {
-    return asText ? displayContent : <Address address={displayContent} />;
+  if (typeof displayContent === "string") {
+    if (isAddress(displayContent)) {
+      return asText ? displayContent : <Address address={displayContent} />;
+    }
+
+    if (isHex(displayContent)) {
+      return displayContent; // don't add quotes
+    }
   }
 
   if (Array.isArray(displayContent)) {
-    const mostReadable = (v: DisplayContent) =>
-      ["number", "boolean"].includes(typeof v) ? v : displayTxResultAsText(v);
-    const displayable = JSON.stringify(displayContent.map(mostReadable), replacer);
-
-    return asText ? (
-      displayable
-    ) : (
-      <span style={{ overflowWrap: "break-word", width: "100%" }}>{displayable.replaceAll(",", ",\n")}</span>
-    );
+    return asText ? JSON.stringify(displayContent, replacer) : <TupleDisplay value={displayContent} />;
+  }
+
+  if (typeof displayContent === "object") {
+    return <StructDisplay value={displayContent} />;
   }
 
   return JSON.stringify(displayContent, replacer, 2);
 };
 
-const displayTxResultAsText = (displayContent: DisplayContent) => displayTxResult(displayContent, true);
+const NumberDisplay = ({ value }: { value: bigint }) => {
+  const [isEther, setIsEther] = useState(false);
+
+  const asNumber = Number(value);
+  if (asNumber <= Number.MAX_SAFE_INTEGER && asNumber >= Number.MIN_SAFE_INTEGER) {
+    return String(value);
+  }
+
+  return (
+    <span>
+      {isEther ? "Ξ" + formatEther(value) : String(value)}
+      <button
+        className="btn btn-primary btn-square btn-xs tooltip tooltip-secondary font-sans ml-2"
+        data-tip={isEther ? "Format as number" : "Format as ether"}
+        onClick={() => setIsEther(!isEther)}
+      >
+        /
+      </button>
+    </span>
+  );
+};
+
+const TupleDisplay = ({ value }: { value: DisplayContent[] }) => {
```

Comment:
Nitpick: Maybe call this as `ArrayDisplay` because in abi representation actually structs are called as tuple 

<details>

<summary>
Example : 
</summary>

![Screenshot 2024-05-29 at 2 41 38 PM](https://github.com/scaffold-eth/scaffold-eth-2/assets/80153681/a44dd351-e419-4885-99a5-e461f8cca931)


</details>

Reply:
Fixed in https://github.com/ByteAtATime/scaffold-eth-2/commit/4a25c0d09d5971bce21bb826ae5d73c09bca7827